cl/donate/api_views.py

```diff
@@ -0,0 +1,215 @@
+                raise NotImplementedError(f"Unknown event trigger-{trigger}")
+
+        return Response(status=HTTPStatus.CREATED)
+
+    def _get_member_record(self, account_id: str) -> User:
+        """
+        Retrieves a user record associated with a Neon CRM account ID.
+
+        This method attempts to find a matching Django user in two ways:
+
+        1. first tries to directly query the database for a user whose
+        `neon_account_id` field matches the provided `account_id`.
+        2. If no matching user is found in the database, it fetches the
+        account email address from the Neon API using the `account_id`.
+        It then tries to find a user whose email address matches the
+        retrieved Neon account email.
+
+        Args:
+            account_id (str): Unique identifier assigned by Neon to an account
+
+        Returns:
+            User: User object associated with the Neon account
+        """
+        try:
+            user = User.objects.get(profile__neon_account_id=account_id)
+        except User.DoesNotExist:
+            client = NeonClient()
+            neon_account = client.get_acount_by_id(account_id)
+            users = User.objects.filter(
+                email=neon_account["primaryContact"]["email1"]
+            )
+            if not users:
+                return HttpResponse(
+                    "Error processing webhook, User not found",
+                    status=HTTPStatus.INTERNAL_SERVER_ERROR,
+                )
+            user = users.first()
+
+            profile = user.profile
+            profile.neon_account_id = neon_account["accountId"]
+            profile.save()
+
+        return user
+
+    def _get_membership_data(
+        self, webhook_data: dict[str, str]
+    ) -> dict[str, str]:
+        """
+        Extracts relevant membership information from a Neon webhook payload.
+
+        Args:
+            webhook_data (dict[str, str]): The payload received from a Neon
+            webhook.
+
+        Returns:
+            dict[str, str]: A dictionary containing the extracted membership
+            data, with the following keys:
+            - membershipId: Unique identifier for the membership.
+            - accountId: Identifier for the user account associated with the
+            membership.
+            - termEndDate: The date the membership is scheduled to terminate
+            - membershipName: The name of the membership level the user is
+            enrolled in.
+        """
+        if "membership" in webhook_data["data"]:
+            data = webhook_data["data"]["membership"]
+            membership = {
+                "membershipId": data["membershipId"],
+                "membershipName": data["membershipName"],
+            }
+        else:
+            data = webhook_data["data"]
+            membership = {
+                "membershipId": data["id"],
+                "membershipName": data["membershipLevel"]["name"],
+            }
```

Comment:
A comment explaining this if/else would be helpful. Why are there two formats of the webhook data?

Reply:
> Why are there two formats of the webhook data?

During initial development, I used the provided [documentation](https://developer.neoncrm.com/api/webhooks/membership-webhooks/#createMembership) to create this helper method. However, upon testing the membership form, I discovered a discrepancy between the expected payload and the actual payload received.

According to the webhook documentation, the `membership` key within the `data` object should provide details about membership changes. However, this key was not present in the webhooks I got by `creating/upgrading/downgrading/renewing` memberships during my testing.

I made this helper compatible with both webhook data formats in case the payload structure outlined in the documentation represents a planned schema update.